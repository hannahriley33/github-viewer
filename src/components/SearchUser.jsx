import React from 'react';
import PropTypes from 'prop-types';

const SearchUser = ({ userName, handleSubmit, handleChange }) => (
  <form onSubmit={handleSubmit}>
    <input type="text" name="userName" value={userName} onChange={handleChange} />
    <button name="userName" value={userName} >Search</button>
  </form>
);
SearchUser.propTypes = {
  userName: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};
export default SearchUser;
