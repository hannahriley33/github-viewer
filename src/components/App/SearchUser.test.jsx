import React from 'react';
import { shallow } from 'enzyme';
import SearchUser from '../SearchUser.jsx';

describe('SearchUser', () => {
  it('displays an input field', () => {
    const wrapper = shallow(<SearchUser  />);
    expect(wrapper).toMatchSnapshot();
  });
});
