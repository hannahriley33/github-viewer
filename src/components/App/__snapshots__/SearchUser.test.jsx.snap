// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchUser displays an input field 1`] = `
<Fragment>
  <input
    name="name"
    type="text"
  />
  <button
    name="button"
    type="button"
  >
    Search for user 
  </button>
</Fragment>
`;
